Group Chats Docs

fabriguespe opened the pull request:
- [Preview](https://junk-range-possible-git-groupchats-xmtp-labs.vercel.app/docs/build/groups)

scotboyd commented on docs/build/groups.md:
```diff
@@ -0,0 +1,382 @@
+
+## Manage group chat members
+
+Here are the ways that you manage group chat members with XMTP.
```
Is this TBD?

scotboyd commented on docs/build/groups.md:
```diff
@@ -0,0 +1,382 @@
+
+### Add group members
+
+Add new members to an existing group chat using its wallet address.
```
```suggestion
Add new members to an existing group chat using their wallet addresses.
```

scotboyd commented on docs/build/groups.md:
```diff
@@ -0,0 +1,382 @@
+
+### Remove group members
+
+Remove a member from an existing group chat using its wallet address
```
```suggestion
Remove members from an existing group chat using their wallet addresses
```

This is based on the title of the section being plural and assuming the function takes more than one.

jhaaaa reviewed (commented):
Thank you for being so ahead of the curve on this, @fabriguespe! =)

I added a few [style and copyedits in this commit](https://github.com/xmtp/xmtp-dot-org/pull/650/commits/365d94e972888a8fc5c2af25bbc721bb603d58ff). Let me know if you have any questions or concerns!

fabriguespe commented:
- Added stream snippets for all SDKs
- Added call out for per-installation group chats (message history)
- Added call out for generic conversations and messages features
- Added sync groups
- Finalized all snippets for group chats
-

nplasterer commented on docs/build/streams.md:
````diff
@@ -327,3 +327,99 @@ for await (const message of await xmtp.conversations.streamAllMessages()) {
+}
+```
+
+Keep your conversation list current by streaming updates for both group and individual conversations using `includeGroups`.
````
```suggestion
Keep your conversation list current by streaming updates for both group and individual conversations.
```

nplasterer commented on docs/build/groups.md:
```diff
@@ -0,0 +1,618 @@
+const group = await client.conversations.newGroup([
+  walletAddress1,
+  walletAddress2,
+]);
```
You could call out the new admin permissions we just shipped as well.

You can now pass a permissions field to newGroup that can take a enum of 
everyone_is_admin
creator_is_admin

Which either allows everyone to add and remove members or only the group creator to do that.

nplasterer commented on docs/build/groups.md:
````diff
@@ -0,0 +1,618 @@
+
+```kotlin
+// List all group chats for the current user
+val groups = client.conversations.listGroups()
````
Might be a good place to call out client.conversations.syncGroups() since the list groups will just pull whatever is in the database. So if you want the latest sync before listing to kick off a network request for the latest groups.

nplasterer commented on docs/build/groups.md:
````diff
@@ -0,0 +1,618 @@
+
+```swift
+try await group.sync()
+let members = group.members()
````
This got renamed to `memberAddresses`
```suggestion
let members = group.memberAddresses()
```
